homeassistant/components/androidtv/config_flow.py

```diff
@@ -0,0 +1,304 @@
+    CONF_ADB_SERVER_IP,
+    CONF_ADB_SERVER_PORT,
+    CONF_ADBKEY,
+    CONF_APPS,
+    CONF_EXCLUDE_UNNAMED_APPS,
+    CONF_GET_SOURCES,
+    CONF_SCREENCAP,
+    CONF_STATE_DETECTION_RULES,
+    CONF_TURN_OFF_COMMAND,
+    CONF_TURN_ON_COMMAND,
+    DEFAULT_ADB_SERVER_PORT,
+    DEFAULT_DEVICE_CLASS,
+    DEFAULT_EXCLUDE_UNNAMED_APPS,
+    DEFAULT_GET_SOURCES,
+    DEFAULT_PORT,
+    DEFAULT_SCREENCAP,
+    DEVICE_CLASSES,
+    DOMAIN,
+)
+
+APPS_NEW_ID = "NewApp"
+CONF_APP_DELETE = "app_delete"
+CONF_APP_ID = "app_id"
+CONF_APP_NAME = "app_name"
+
+RESULT_CONN_ERROR = "cannot_connect"
+RESULT_UNKNOWN = "unknown"
+RESULT_SUCCESS = "success"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _is_file(value):
+    """Validate that the value is an existing file."""
+    file_in = os.path.expanduser(str(value))
+
+    return os.path.isfile(file_in) and os.access(file_in, os.R_OK)
+
+
+def _get_ip(host):
+    """Get the ip address from the host name."""
+    try:
+        return socket.gethostbyname(host)
+    except socket.gaierror:
+        return None
+
+
+class AndroidTVFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize Android TV config flow."""
+        self._host = None
+        self._unique_id = None
+
+    @callback
+    def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+                    vol.Required(
+                        CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS
+                    ): vol.In(DEVICE_CLASSES),
+                    vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                    vol.Optional(CONF_ADBKEY): str,
+                    vol.Optional(CONF_ADB_SERVER_IP): str,
+                    vol.Required(
+                        CONF_ADB_SERVER_PORT, default=DEFAULT_ADB_SERVER_PORT
+                    ): cv.port,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def _async_check_connection(self, user_input):
+        """Attempt to connect the Android TV."""
+
+        try:
+            aftv = await async_connect_androidtv(self.hass, user_input, timeout=30.0)
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Unknown error connecting with Android TV at %s", self._host
+            )
+            return RESULT_UNKNOWN
+
+        if not aftv:
+            return RESULT_CONN_ERROR
+
+        dev_prop = aftv.device_properties
+        self._unique_id = dev_prop.get("ethmac") or dev_prop.get("wifimac")
+        await aftv.adb_close()
+        return RESULT_SUCCESS
```

Comment:
```suggestion
                "Unknown error connecting with Android TV at %s", user_input[CONF_HOST]
            )
            return RESULT_UNKNOWN, None

        if not aftv:
            return RESULT_CONN_ERROR, None

        dev_prop = aftv.device_properties
        unique_id = format_mac(dev_prop.get("ethmac", "wifimac"))
        await aftv.adb_close()
        return None, unique_id
```

Reply:
```
       unique_id = format_mac(dev_prop.get("ethmac", "wifimac"))
```

This is wrong, we want to get  the item with ID "wifimac" if the one with "ethmac" is not available, not set "wifimac" as default